common/src/main/java/com/wynntils/features/combat/QuickCastFeature.java

```diff
@@ -160,8 +163,8 @@ public void onTick(TickEvent e) {
             spellInProgress = SpellDirection.NO_SPELL;
         }
 
+        if (packetCountdown > 0 && --packetCountdown > 0) return;
```

Comment:
I see this is merged now, and that it was almost this bad before, but this is really not an ideal piece of code.

Using ++ or -- should nearly always be done on a separate line. And having side effects in an if condition is nearly always bad. 

```
if (packetCountdown > 0) {
   packetCountdown--;
}
if (packetCountdown > 0) return;
```

would have been much clearer.

Reply:
now I know. Maybe I'll make a pr to extract side effects from all ifs